sqldelight-idea-plugin/src/main/kotlin/com/squareup/sqldelight/intellij/intentions/IntroduceTableAliasIntention.kt

```diff
@@ -0,0 +1,108 @@
+package com.squareup.sqldelight.intellij.intentions
+
+import com.alecstrong.sql.psi.core.psi.SqlColumnName
+import com.alecstrong.sql.psi.core.psi.SqlCreateTableStmt
+import com.alecstrong.sql.psi.core.psi.SqlSelectStmt
+import com.alecstrong.sql.psi.core.psi.SqlTableAlias
+import com.alecstrong.sql.psi.core.psi.SqlTableName
+import com.intellij.codeInsight.intention.BaseElementAtCaretIntentionAction
+import com.intellij.openapi.command.WriteCommandAction
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.popup.JBPopupFactory
+import com.intellij.psi.PsiElement
+import com.intellij.psi.util.PsiTreeUtil
+import com.intellij.psi.util.parentOfType
+import com.squareup.sqldelight.core.lang.util.findChildOfType
+import com.squareup.sqldelight.core.lang.util.findChildrenOfType
+
+class IntroduceTableAliasIntention : BaseElementAtCaretIntentionAction() {
+
+  private val regex = "(?=\\p{Upper})|(_)".toRegex()
+
+  override fun getFamilyName(): String {
+    return INTENTIONS_FAMILY_NAME_REFACTORINGS
+  }
+
+  override fun getText(): String {
+    return "Introduce table alias"
+  }
+
+  override fun isAvailable(project: Project, editor: Editor, element: PsiElement): Boolean {
+    val tableName = element.parentOfType<SqlTableName>(true) ?: return false
+    val columnAlias = PsiTreeUtil.getNextSiblingOfType(tableName, SqlTableAlias::class.java)
+    if (columnAlias != null) {
+      return false
+    }
+    val sqlSelectStmt = tableName.parentOfType<SqlSelectStmt>() ?: return false
+    return sqlSelectStmt.resultColumnList.none { it.textMatches("*") }
+  }
+
+  override fun invoke(project: Project, editor: Editor, element: PsiElement) {
+    val sqlTableName = element.parentOfType<SqlTableName>(true) ?: return
+    val sqlSelectStmt = sqlTableName.parentOfType<SqlSelectStmt>() ?: return
+
+    val columnDef = sqlTableName.reference?.resolve()
+      ?.parentOfType<SqlCreateTableStmt>()
+      ?.columnDefList
+      .orEmpty()
+      .mapTo(mutableSetOf()) { columnDef -> columnDef.columnName }
```

Comment:
is this also what ultimate does? Personally I would expect us to not add the alias to columns that weren't already using the table name, and those cases would be covered by `tableReferenceMarkers`. If we're okay with that it would shorten this implementation too to not needing `columnDef` or `columns`, which I suspect would mean this would also work for giving an alias to views and common table expressions

Reply:
> If we're okay with that

yup we can add another one like IDEA does
![frame_21_delay-0 4s](https://user-images.githubusercontent.com/14963604/117868111-e1a38180-b2a1-11eb-80a8-0422aa92059c.gif)